robot-core/src/main/java/org/obolibrary/robot/MeasureOperation.java

```diff
@@ -0,0 +1,330 @@
+package org.obolibrary.robot;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
+import com.google.gson.*;
+import java.io.*;
+import java.util.*;
+import org.obolibrary.robot.export.Cell;
+import org.obolibrary.robot.export.Column;
+import org.obolibrary.robot.export.Row;
+import org.obolibrary.robot.export.Table;
+import org.obolibrary.robot.metrics.MeasureResult;
+import org.obolibrary.robot.metrics.OntologyMetrics;
+import org.obolibrary.robot.providers.CURIEShortFormProvider;
+import org.semanticweb.owlapi.model.OWLOntology;
+import org.semanticweb.owlapi.reasoner.OWLReasoner;
+import org.semanticweb.owlapi.reasoner.OWLReasonerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Compute metrics for the ontology.
+ *
+ * @author <a href="mailto:nicolas.matentzoglu@gmail.com">Nicolas Matentzoglu</a>
+ */
+public class MeasureOperation {
+  /** Logger. */
+  private static final Logger logger = LoggerFactory.getLogger(MeasureOperation.class);
+
+  /** Namespace for error messages. */
+  private static final String NS = "measure#";
+
+  /** Error message when metric type is illegal. Expects: metric type. */
+  private static final String metricsTypeError = NS + "METRICS TYPE ERROR unknown metrics type: %s";
+
+  /** Error message when format type is illegal. Expects: format. */
+  private static final String metricsFormatError =
+      NS + "METRICS FORMAT ERROR unknown metrics format: %s";
+
+  /**
+   * If a result set has results, write to the output stream and return true. Otherwise return
+   * false.
+   *
+   * @param result the results to write
+   * @param format the name of the file format to write the results to
+   * @param output the file to write to
+   * @return true if there were results, false otherwise
+   * @throws IOException if writing file failed
+   */
+  public static boolean maybeWriteResult(MeasureResult result, String format, File output)
+      throws IOException {
+    if (!result.isEmpty()) {
+      writeResult(result, format, output);
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public static void executeMetrics(
+      OWLOntology o,
```

Comment:
This is super minor and I'm not sure if it's important, but the `OWLOntology` is called `ontology` everywhere else (`OWLOntology ontology`).